### PR TITLE
perf improvement attempts for OpenGL.c

### PR DESCRIPTION
**Pull request by hilga007**
avoiding unnecessary memory allocations and string copying

### CODE REVIEW

File: src/modules/opengl.c [hunk 1]
```diff
@@ -7,36 +7,29 @@
 
 void ffPrintOpenGL(FFinstance* instance)
 {
-    FFOpenGLResult result;
-    ffStrbufInit(&result.version);
-    ffStrbufInit(&result.renderer);
-    ffStrbufInit(&result.vendor);
-    ffStrbufInit(&result.slv);
```
Comment (LinusDierheimer) [inline src/modules/opengl.c:14]: If the strbufs aren't initialized, `ffDetectOpenGL` will write data in random memory places, which most probably will cause a seg fault

File: src/modules/opengl.c [hunk 2]
```diff
@@ -7,36 +7,29 @@
...
+            { FF_FORMAT_ARG_TYPE_STRBUF, &result.renderer },
+            { FF_FORMAT_ARG_TYPE_STRBUF, &result.vendor },
+            { FF_FORMAT_ARG_TYPE_STRBUF, &result.slv }
+        };
 
-    ffStrbufDestroy(&result.version);
-    ffStrbufDestroy(&result.renderer);
-    ffStrbufDestroy(&result.vendor);
-    ffStrbufDestroy(&result.slv);
+        ffPrintFormat(instance, FF_OPENGL_MODULE_NAME, 0, &instance->config.openGL, FF_OPENGL_NUM_FORMAT_ARGS, formatArgs);
```
Comment (LinusDierheimer) [inline src/modules/opengl.c:33]: Why? The compiled assembly will be the same

File: src/modules/opengl.c (hunk 1 above)
Comment (LinusDierheimer) [reply src/modules/opengl.c:14]: The reason that the pipeline doesn't fail is that the GitHub VMs don't have OpenGL support

File: src/modules/opengl.c (hunk 2 above)
Comment (hilga007) [reply src/modules/opengl.c:33]: Again - did not mean to submit to you - the "why" is just "this is what I have been learning w/o a class and w/o a university and wanted to apply a small change based on best practice" 

I guess so maybe what you are saying is that "it does not matter which path is taken, so long as the destination is reached?" Whereas that is the opposite of what I had bothered to learn, which is why this change had been made. 

"Conformance is performance" was one of the small sayings.

Comment (LinusDierheimer) [reply src/modules/opengl.c:33]: The example is similar to the following:
```c
void do_something(int);

//My Version
do_something(5);

//Your version
int a = 5;
do_something(a);
```
Compiled they are the same, so lets only talk about the code style. Mine is more simple, and still readable, so it is the better solution here. (In my opionion, we are at a point where personal opionion does make the difference)

Comment (hilga007) [reply src/modules/opengl.c:33]: And experience! Your experience makes it make sense! Thank you!!! As they say "real world experience" makes a difference over "textbook experience" :) :) 

Sorry for the panic attack..... 
Again I am very new with this. Only reading C99 books when young, and then 2 months of a course + textbook.

**Pull request closed without merging**